common/djangoapps/nodebb/tasks.py

```diff
@@ -38,18 +38,13 @@ def handle_response(caller, task_name, status_code, response, username=None):
 
 
 @task(default_retry_delay=RETRY_DELAY, max_retries=None, routing_key=settings.HIGH_PRIORITY_QUEUE)
-def task_create_user_on_nodebb(username, user_data):
+def task_create_user_on_nodebb(username, user_data, user):
```

Comment:
I think we should only pass `user ` and get username from user inside this method instead of passing it where it gets called.